### PR TITLE
Add correct checking for whether a command is supported.

### LINKED ISSUE #7612: Refactor the error reporting when the path is broken or target not exists
This doesn't seem right.  We could be here because the path is broken (INVALID_COMMAND, I think) or the path target does not exist (UNSUPPORTED_ENDPOINT, UNSUPPORTED_CLUSTER, UNSUPPORTED_COMMAND).

Please file a followup to do better error reporting here?

_Originally posted by @bzbarsky-apple in https://github.com/project-chip/connectedhomeip/pull/7482#discussion_r649582430_

### LINKED ISSUE #11185: [Test-TC-DIASW-1.1] ThreadMetrics attribute seems to be missing from the Software Diagnostics cluster
#### Problem
The optional attribute ThreadMetrics seems to be missing from the  Software Diagnostics cluster.

**Spec Ref**
SoftwareDiagnosticsAttributes

**YAML Test Script Error**
TC-DIAGSW-1 1

### LINKED ISSUE #11578: [TC-DIAGSW-3.2] Command failing --ResetWatermarks
Summary :
As per test plane the command  ResetWatermarks is  to be send from the TH and should return a successful status, however when we try to use this command in the YAML test scripts they seem to always fail, please refer attached screenshots and logs

Test-plan Screenshot:
image

Spec Screenshot:
image

Log file and Execution Screenshot:
[DIASW_3_2.txt](https://github.com/project-chip/connectedhomeip/files/7503579/DIASW_3_2.txt)
image

### LINKED ISSUE #13920: in particular different reasons for ServerClusterCommandExists to test false should result in different errors here
false

---

https://github.com/project-chip/connectedhomeip/blob/f0f5de0d79ae1c51d7fddbf70900ebac0acd899a/src/app/CommandHandler.cpp#L306-L311

---

###### This issue was generated by [todo-issue](https://github.com/DerJuulsn/todo-issue) based on a `TODO` comment in f0f5de0d79ae1c51d7fddbf70900ebac0acd899a.

### LINKED ISSUE #14022: Replace implementation of `ServerClusterCommandExists()` in ember-compatibility-functions.cpp by IM command catalog look-up
false

---

https://github.com/project-chip/connectedhomeip/blob/f0f5de0d79ae1c51d7fddbf70900ebac0acd899a/src/app/InteractionModelEngine.h#L264-L269

---

###### This issue was generated by [todo-issue](https://github.com/DerJuulsn/todo-issue) based on a `TODO` comment in f0f5de0d79ae1c51d7fddbf70900ebac0acd899a.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
This also fixes the status reporting for unsupported endpoints and clusters.

Fixes https://github.com/project-chip/connectedhomeip/issues/7612
Fixes https://github.com/project-chip/connectedhomeip/issues/13920
Fixes https://github.com/project-chip/connectedhomeip/issues/14022
Fixes https://github.com/project-chip/connectedhomeip/issues/11578
Fixes https://github.com/project-chip/connectedhomeip/issues/11185

#### Problem
1. We are not correctly rejecting commands that are supported on the given cluster on some other endpoint but not on the targeted one.
2. We are not reporting the right status for unsupported commands.

#### Change overview
Fix those issues by using the ClientGeneratedCommandList metadata we have.

Also fix the status bits for attributes.

#### Testing
Modified existing yaml tests to test the new status.

Modified existing yaml tests that were failing now due to trying to talk to the wrong endpoint.

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

File: examples/bridge-app/esp32/main/main.cpp [hunk 1]
```diff
@@ -97,10 +97,23 @@ DECLARE_DYNAMIC_ATTRIBUTE(ZCL_LABEL_LIST_ATTRIBUTE_ID, ARRAY, kFixedLabelAttribu
     DECLARE_DYNAMIC_ATTRIBUTE_LIST_END();
 
 // Declare Cluster List for Bridged Light endpoint
+// TODO: It's not clear whether it would be better to get the command lists from
+// the ZAP config on our last fixed endpoint instead.
```
Comment (mrjerryjohns) [inline examples/bridge-app/esp32/main/main.cpp:101]: But that's a grab-bag of all possible commands spread across all endpoints right?

File: examples/tv-app/linux/AppImpl.cpp [hunk 2]
```diff
@@ -132,17 +132,94 @@ DECLARE_DYNAMIC_ATTRIBUTE(ZCL_CHANNEL_LIST_ATTRIBUTE_ID, ARRAY, kDescriptorAttri
     DECLARE_DYNAMIC_ATTRIBUTE(ZCL_CHANNEL_CURRENT_CHANNEL_ATTRIBUTE_ID, STRUCT, 1, 0),             /* current channel */
     DECLARE_DYNAMIC_ATTRIBUTE_LIST_END();
 
+constexpr CommandId keypadInputIncomingCommands[] = {
```
Comment (mrjerryjohns) [inline examples/tv-app/linux/AppImpl.cpp:135]: This is unfortunate that we've ended up in this place, where the application author has to carefully create this list to match what the cluster impl expects.

The cluster impl **is** the authority on knowing which commands it should/should not expect. If there are feature flags that alter that behavior, those should be protected with compile/dynamic feature flags (if any).

Anyways...

**Review by mrjerryjohns — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

File: examples/bridge-app/esp32/main/main.cpp (hunk 1 above)
Comment (bzbarsky-apple) [reply examples/bridge-app/esp32/main/main.cpp:101]: Right, hence not clear.

File: examples/tv-app/linux/AppImpl.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply examples/tv-app/linux/AppImpl.cpp:135]: The cluster impl might be able to handle commands A, B, C, but the application only wants to enable commands A, B on endpoint 1 and commands B, C on endpoint 2.  The application is the only thing that knows which commands are actually enabled on a given endpoint.

In ZAP, this is accomplished by selecting things in the ZAP UI.  When using external endpoints not defined via ZAP, it has to be defined somehow, by the application.

**Pull request merged**